### PR TITLE
Restrict authority to lowercase unreserved characters only

### PR DESCRIPTION
**Pull request by sophokles73**

### CODE REVIEW

**Review by stevenhartley — APPROVED** [review]
LGTM

**Comment by sophokles73** [comment]
@evshary does this work for you as well?

File: basics/uri.adoc [hunk 1]
```diff
@@ -133,16 +133,19 @@ URI = "up" ":" hier-part
...
-[.specitem,oft-sid="dsn~uri-host-only~1",oft-needs="impl,utest"]
---
-The _authority_ component of a URI *MUST* consist of a _host_ part only, i.e. it *MUST NOT* contain neither _userinfo_ nor _port_ information.
+[.specitem,oft-sid="dsn~uri-host-only~2",oft-needs="impl,utest"]
 --
+The _authority_ component of a URI *MUST* consist of characters according to the following rule:
 [source,abnf]
 ----
-authority = host
+lc-unreserved = *( %x61-7A / DIGIT / "-" / "." / "_" / "~" )
```
Comment (evshary) [inline basics/uri.adoc:141]: So we don't allow URL encoding now, right?
For example: can we use `//%78%79%7A/...`  instead of `//xyz/...`?

**Review by evshary — APPROVED** [review]
I like this change, which makes our life easier.
Only one question to be confirmed, otherwise it looks perfect to me.

File: basics/uri.adoc (hunk 1 above)
Comment (sophokles73) [reply basics/uri.adoc:141]: No, there is no more percent encoding. The reason why we wanted to remove support for it is, that in order to properly match filter URIs (for invoking listeners), we would always need to _normalize_ all URIs, which might be quite expensive. Is that an issue?

Comment (evshary) [reply basics/uri.adoc:141]: No. Actually, it's better for me. :laughing:

**Pull request merged**